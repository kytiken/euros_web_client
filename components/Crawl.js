import React from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import Link from 'next/link';
import { connect } from 'react-redux';

const Crawl = ({
  crawl,
}) =>
  (
    <li>
      <Link href={{ pathname: '/desk', query: { crawlId: crawl.id } }}>
        <a>{ crawl.url }</a>
      </Link>
    </li>
  );


Crawl.propTypes = {
  crawl: ImmutablePropTypes.recordOf({ url: PropTypes.string }).isRequired,
};

export default connect(state => state)(Crawl);
