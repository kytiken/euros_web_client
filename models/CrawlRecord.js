import { Record } from 'immutable';

const crawlAttr = {
  id: null,
  url: '',
  depthLimit: null,
  timeout: null,
  recvTimeout: null,
  pattern: '',
  cookie: '',
};

export default class CrawlRecord extends Record(crawlAttr) {
}
