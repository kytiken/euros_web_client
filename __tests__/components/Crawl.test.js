import React from 'react';
import renderer from 'react-test-renderer';
import Crawl from '../../components/Crawl';
import CrawlRecord from '../../models/CrawlRecord';

it('renders correctly', () => {
  const crawl = new CrawlRecord({
    id: 1,
    url: 'http://google.com',
    depthLimit: 2,
    timeout: 60000,
    recvTimeout: 70000,
    pattern: '.*',
    cookie: 'hoge',
  });
  const tree = renderer
    .create(<Crawl key={crawl.id} crawl={crawl} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
