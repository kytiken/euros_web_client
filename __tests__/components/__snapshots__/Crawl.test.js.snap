// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <a
    href="/desk?crawlId=1"
    onClick={[Function]}
  >
    http://google.com
  </a>
</li>
`;
